runtime/intern.c

```diff
@@ -439,6 +439,15 @@ static value intern_alloc_obj(struct caml_intern_state* s, caml_domain_state* d,
     }
     d->allocated_words += Whsize_wosize(wosize);
     d->allocated_words_direct += Whsize_wosize(wosize);
+    /* #13300: We assume that marshalled data whose total payload size
+       is above Max_young_wosize (2Mio on 64 machines) is long-lived.
```

Comment:
`Max_young_wosize` is 256 on all machines, regardless of minor heap size (see https://github.com/ocaml/ocaml/blob/cea4b6d84161e3d35eb6966bdbfaa225de44832c/runtime/caml/config.h#L202). It's probably too small for this purpose.

Reply:
256 words is 256*8 = 2MiB (in bytes) on 64 machines, right?

What other cutoff would you suggest? (Should we make this a settable config parameter?)

Reply:
Duh, 256*8 is 2Kib, not 2Mib.